src/github/utils/workflow-dispatch.ts

```diff
@@ -32,6 +32,17 @@ export async function dispatchWorkflow(context: GitHubContext, options: Workflow
   });
 }
 
+export async function dispatchWorker(targetUrl: string, payload: Record<string, unknown>) {
+  const result = await fetch(targetUrl, {
+    body: JSON.stringify(payload),
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  return result.json();
```

Comment:
I suppose worker plugins will be only used for short-running tasks so it's probably fine to get results back directly from the response instead of having a callback

Reply:
Maybe it is better to still rely on a callback to avoid changing the whole logic just for workers? I don't know how much changes this implies in the codebase.

Reply:
We could make a callback but it needs to secure, one idea I posted in the issue was to generate a UUID every time you call a worker plugin and store it in the KV, and in the callback we can validate that state id and invocation id match

Reply:
Agree. I was thinking also Workers can use `repository_dispatch` since they have access to Octokit, which would keep the logic the same for both and still be instantaneous but I guess it would not make much sense for plugins that do not require an instance of Octokit, although it might be rare. Will look into it.

Reply:
First idea: worker plugins can return a `204` when they don't have any data to return and nothing to dispatch, or return `200` thus triggering the dispatch. Afterwards we can return to the normal [repository dispatch](https://github.com/ubiquity/ubiquibot-kernel/blob/98ca6744bbe62b22fade179149c7015697c73f1c/src/github/handlers/repository-dispatch.ts#L7) flow. If we pick the result directly from the [result of the fetch](https://github.com/ubiquity/ubiquibot-kernel/blob/98ca6744bbe62b22fade179149c7015697c73f1c/src/github/utils/workflow-dispatch.ts#L43) that would save us the need to secure it with the UUID.

Reply:
@whilefoo I think this might fall out of scope of this PR, because their is a wider security concern on the worker themselves and how to spawn them securely. I opened another issue on that regard, feel free to add your thoughts there as well: https://github.com/ubiquity/ubiquibot-kernel/issues/43